charms/reactive/bus.py

```diff
@@ -40,21 +40,31 @@ def __init__(self, path):
         super(BrokenHandlerException, self).__init__(message)
 
 
-def _action_id(action):
+def _action_id(action, suffix=None):
     if hasattr(action, '_action_id'):
         return action._action_id
-    return "%s:%s:%s" % (action.__code__.co_filename,
-                         action.__code__.co_firstlineno,
-                         action.__code__.co_name)
+    parts = [
+        action.__code__.co_filename,
+        action.__code__.co_firstlineno,
+        action.__code__.co_name,
+    ]
+    if suffix is not None:
+        parts.append(suffix)
+    return ':'.join(map(str, parts))
 
 
-def _short_action_id(action):
+def _short_action_id(action, suffix=None):
```

Comment:
Should suffix be optional? Its an internal method, so making it a required argument will help catch any call sites we missed.